### DISCUSSION TITLE
Try to use vector_based_blocking.py

### DISCUSSION THREAD

**Discussion by NguyenQS** [AI-team-UoA/pyJedAI#6] [Q&A] [answered]
Hi,

great work by the way. I have a bit problems to understand and use the 'vector_based_blocking.py'. I do no know how to integrate it in the workflow. It only creates blocks like block_building + comparison_cleaning.py does, am i right? and directly after that i can apply the 'matching.py'-code without an intermediate step and use metrics like jaccard or is it not possible to use jaccard with 'word2vec' for example?

Or where are the differences between this code 'block_building.py' with regard to the architecture and workflow?

I look forward for a fast answer and thank you in advance

**Comment by Nikoletos-K** [comment, marked as answer]
Hello!
'vector based blocking' module generates blocks with the same structure as the 'block building' module. You can evaluate them similarly to how you would evaluate a block building method by utilizing the Evaluate class. Yes, you can utilize the matching step immediately after performing 'vector based blocking'. The primary difference between these two architectures is that they take entirely different approaches. In the first scenario, the block building module employs a technique that makes use of the entities' tokens, whereas 'vector based blocking' use pre-trained embeddings to transform the strings and then queries the embeddings using FAISS. Be aware that SCANN and FALCONN are not yet implemented in this version of pyJedAI, but we hope that they will be in the next release.

Thank you for your interest in pyJedAI, and please do not hesitate to contact me if you have any more questions.
